### PR TITLE
Chore: Update Thanos

### PR DESCRIPTION
**Pull request by harry671003**
**What this PR does**:
- Update Thanos to `09db52562de0984d097554387274a654db19ce83`

```
> go get github.com/thanos-io/thanos@09db52562de0984d097554387274a654db19ce83
> go mod tidy
> go mod vendor
```

- Note: tests were breaking due to https://github.com/prometheus/client_golang/pull/1424

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by yeya24** [comment]
Is the data race relevant to this change? Looks a bit concerning

**Comment by harry671003** [comment]
@yeya24 - The data race seems to have gone away in the latest change.

File: pkg/ruler/user2/namespace
Comment (yeya24) [inline pkg/ruler/user2/namespace:1]: Those files are used for testing only and should be excluded.

File: pkg/compactor/compactor_test.go [hunk 1]
```diff
@@ -263,26 +263,12 @@ func TestCompactor_ShouldDoNothingOnNoUserBlocks(t *testing.T) {
...
-		"cortex_compactor_group_compactions_total",
-		"cortex_compactor_group_vertical_compactions_total",
 		"cortex_compactor_block_cleanup_failures_total",
 		"cortex_compactor_blocks_cleaned_total",
-		"cortex_compactor_blocks_marked_for_deletion_total",
 		"cortex_compactor_blocks_marked_for_no_compaction_total",
+		"cortex_compactor_meta_sync_consistency_delay_seconds",
 		"cortex_compactor_block_cleanup_started_total",
 		"cortex_compactor_block_cleanup_completed_total",
-		"cortex_compactor_block_cleanup_failed_total",
```
Comment (yeya24) [inline pkg/compactor/compactor_test.go:285]: Why those metrics are removed?

**Review by yeya24 — APPROVED** [review]
Thanks!

**Pull request merged**